### PR TITLE
Improve iot_uart.h with enum

### PR DESCRIPTION
**Pull request by haipengd**
Description
Improve iot_uart.h with enum
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by haipengd** [comment]
@vjjunnut @qiutongs can you help to review

File: libraries/abstractions/common_io/test/test_iot_uart.c [hunk 1]
```diff
@@ -78,10 +78,10 @@ static StaticSemaphore_t xtestIotUARTCompleted;
...
 {
     pxConfig->ulBaudrate   = testIotUART_BAUD_RATE_FAST;
-    pxConfig->ulFlowControl= 0;
-    pxConfig->ulParity     = 0;
-    pxConfig->ulStopbits   = 0;
-    pxConfig->ulWordlength = 3;
+    pxConfig->ucFlowControl= 0;
+    pxConfig->xParity     = 0;
+    pxConfig->xStopbits   = 0;
+    pxConfig->ucWordlength = 3;
```
Comment (qiutongs) [inline libraries/abstractions/common_io/test/test_iot_uart.c]: Better to be:
```
xParity = eUartParityNone
xStopbits = eUartStopBitsOne
```

**Comment by haipengd** [comment]
@qiutongs make sense. Will update it.

**Comment by haipengd** [comment]
@qiutongs: I pushed a change to address your comment. Can you help to review it?

**Pull request merged**